app/scripts/models/reliers/relier.js

```diff
@@ -13,31 +13,62 @@
 define(function (require, exports, module) {
   'use strict';
 
+  var AuthErrors = require('lib/auth-errors');
   var BaseRelier = require('models/reliers/base');
   var Cocktail = require('cocktail');
   var Constants = require('lib/constants');
   var p = require('lib/promise');
   var ResumeTokenMixin = require('models/mixins/resume-token');
   var SearchParamMixin = require('models/mixins/search-param');
+  var ValidateMixin = require('models/mixins/validate');
+  var Vat = require('vat');
 
   var RELIER_FIELDS_IN_RESUME_TOKEN = [
-    'utmTerm',
-    'utmSource',
-    'utmMedium',
-    'utmContent',
-    'utmCampaign',
     'campaign',
-    'entrypoint'
+    'entrypoint',
+    'utmCampaign',
+    'utmContent',
+    'utmMedium',
+    'utmSource',
+    'utmTerm'
   ];
 
+  /*eslint-disable camelcase*/
+  var QUERY_PARAMETER_SCHEMA = {
+    campaign: Vat.string(),
+    // `email` will be further validated by views to
+    // show the appropriate error message
+    email: Vat.string().allow(Constants.DISALLOW_CACHED_CREDENTIALS),
+    // FxDesktop declares both `entryPoint` (capital P) and
+    // `entrypoint` (lowcase p). Normalize to `entrypoint`.
+    entryPoint: Vat.string(),
+    entrypoint: Vat.string(),
+    migration: Vat.string().valid(Constants.AMO_MIGRATION).valid(Constants.SYNC11_MIGRATION),
```

Comment:
Think `valid` should be able to take more than one argument at a time to avoid repeated calls to `valid`?